### PR TITLE
doc: Update outdated hyperlinks in website

### LINKED ISSUE #109: Broken link in branch protection
Link to [`branch_protection.go`](https://github.com/kubernetes/test-infra/tree/master/prow/config/branch_protection.go) is broken in https://docs.prow.k8s.io/docs/components/optional/branchprotector/

### PR DESCRIPTION
**Pull request by jihoon-seo**
- Update links that point to `k/t-i` repo to point `k-sigs/prow` instead
  (Note: Not all links that point to `k/t-i` repo have been updated, since many docs and codes are still in `k/t-i` repo.)
- Update links that point to `*.md` files to Prow Doc website pages, if exist
- http://github.com/**/tree/**/some.file → http://github.com/**/blob/**/some.file
  (this will remove unnecessary redirections on users' browser)

Before opening this PR, I checked whether links are OK using [`check-broken-links.sh`](https://github.com/kubernetes-sigs/prow/blob/main/site/check-broken-links.sh).

(This PR Fixes #109.)

### CODE REVIEW

File: site/content/en/docs/components/_index.md
Comment (jihoon-seo) [inline site/content/en/docs/components/_index.md:48]: The source code of phaino does not exist in either 
- https://github.com/kubernetes/test-infra/tree/master/prow/cmd/phaino or 
- https://github.com/kubernetes-sigs/prow/tree/main/prow/cmd/phaino

File: site/content/en/docs/components/optional/status-reconciler.md
Comment (jihoon-seo) [inline site/content/en/docs/components/optional/status-reconciler.md:26]: The source code of migratestatus does not exist in either
- https://github.com/kubernetes/test-infra/tree/master/maintenance/migratestatus or
- https://github.com/kubernetes-sigs/prow/tree/main/maintenance/migratestatus

**Review by jihoon-seo — COMMENTED** [review]
For reviewers' information:

File: site/content/en/docs/components/optional/status-reconciler.md
Comment (cjwagner) [reply site/content/en/docs/components/optional/status-reconciler.md:26]: `migratestatus` was removed in this PR: https://github.com/kubernetes/test-infra/pull/32215
We can delete all references to it in the docs.

File: site/content/en/docs/components/_index.md
Comment (cjwagner) [reply site/content/en/docs/components/_index.md:48]: `phaino` was removed in this PR: https://github.com/kubernetes/test-infra/pull/32312
We can delete all references to it in the docs.

**Review by cjwagner — COMMENTED** [review]
Thanks! LGTM, besides removing references to the deleted tools.

File: site/content/en/docs/build-test-update.md
Comment (jihoon-seo) [inline site/content/en/docs/build-test-update.md:145]: I am not sure whether we should delete this paragraph (L145-149) too, or not.

```diff

##### Debugging within a pj-on-kind.sh container
To point `kubectl` to the Kind cluster you will need to export the `KUBECONFIG` Env. The command to point this to the correct config is echoed in the pj-on-kind.sh logging. It will have the form:
(SOME CODEBLOCK)
After pointing to the correct master you will be able to drop into the container using `kubectl exec -it <pod name> <bash/sh/etc>`. **This pod will only last the lifecycle of the job, if you need more time to debug you might add a `sleep` within the job execution.

-Note: Test containers designed for decorated jobs (configured with `decorate: true`)
-may behave incorrectly or fail entirely without the environment the pod utilities
-provide. Similarly jobs that mount volumes or use `extra_refs` likely won't work
-properly.
-These jobs are best run locally as decorated pods inside a [Kind] cluster [Using pj-on-kind.sh](#using-pj-on-kindsh).

### How to manually run a given job on Prow

```

Any comment would be appreciated.

File: site/content/en/docs/components/optional/status-reconciler.md
Comment (jihoon-seo) [inline site/content/en/docs/components/optional/status-reconciler.md:26]: Please review whether this removal is appropriate as-is, or it would be great to remove L24 too.

**Review by jihoon-seo — COMMENTED** [review]
@cjwagner Thank you for your comment!
I reflected your suggestions.
PTAL!

/label tide/merge-method-squash
/retest-required

File: site/content/en/docs/components/deprecated/phaino.md
Comment (jihoon-seo) [inline site/content/en/docs/components/deprecated/phaino.md:1]: FYI: Moved content/en/docs/components/**cli-tools**/phaino.md → content/en/docs/components/**deprecated**/phaino.md,
and updated links that point to `phaino` page too

File: site/content/en/docs/metadata-artifacts.md
Comment (jihoon-seo) [inline site/content/en/docs/metadata-artifacts.md:91]: For reviewers' information:
the link points to `func refsToStarted()`,
and the function was moved from `prow/initupload/run.go` to `prow/pod-utils/downwardapi/jobspec.go` in https://github.com/kubernetes/test-infra/pull/27259.

File: site/content/en/docs/build-test-update.md
Comment (cjwagner) [reply site/content/en/docs/build-test-update.md:145]: Yes, this is part of the "Using Phaino" section and only applies to Phaino. Please remove this as well.

File: site/content/en/docs/components/optional/status-reconciler.md
Comment (cjwagner) [reply site/content/en/docs/components/optional/status-reconciler.md:26]: I think this removal seems appropriate as is. Please leave L24.

File: prow/cmd/checkconfig/main.go
Comment (jihoon-seo) [inline prow/cmd/checkconfig/main.go:520]: This modification is causing `pull-prow-unit-test` to fail. ([error log](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_prow/103/pull-prow-unit-test/))

I guess that the `checkconfig` is intended to accept GitHub blob or tree URLs of `kubernetes`/`k8s` org,
and this causes error because I am trying to replace with `docs.prow.k8s.io` URL, which is not a GitHub URL.

We might
1. Modify `checkconfig` code (or its test code) to accept `docs.prow.k8s.io` URLs as well
   (but I guess that this is against the original intention of `checkconfig`, so not appropriate)
2. Or, update URLs in L520 & L980 to https://github.com/kubernetes-sigs/prow/blob/main/site/content/en/docs/components/core/tide/maintainers.md#best-practices
   (We might need to modify `checkconfig` code to accept `kubernetes-sigs` GitHub org as well.)

**Comment by jihoon-seo** [comment]
The failure of `pull-prow-integration` & `pull-prow-image-build-test` seems to be related with
- https://github.com/kubernetes-sigs/prow/pull/101#issuecomment-2050263757
- https://github.com/kubernetes/test-infra/pull/32433

I guess that after #101 gets merged, a rebasing will be required for this PR.

**Comment by jihoon-seo** [comment]
I performed rebasing, reflecting code file path changes caused by #101.

Quoting @petr-muller's [comment](https://github.com/kubernetes-sigs/prow/pull/106#issuecomment-2061081779):
> Let's wait a day or two though, for the case #101 needs to be reverted, so that we don't make it harder if needed.

File: cmd/checkconfig/main.go
Comment (petr-muller) [inline cmd/checkconfig/main.go:980]: https://docs.prow.k8s.io/docs/components/core/tide/maintainers/#best-practices works, IMO no need to migrate it

File: site/content/en/docs/components/_index.md
Comment (petr-muller) [inline site/content/en/docs/components/_index.md:8]: I'm not a fan of linebreaks in markdown, they make diffs messy. I'd prefer to not do this change.

File: cmd/checkconfig/main.go
Comment (jihoon-seo) [reply cmd/checkconfig/main.go]: I reverted to https://docs.prow.k8s.io/docs/components/core/tide/maintainers/#best-practices in 29341e9, but the reverting caused unit test error..
I might drop the commit 29341e9 and push again. Does this seem ok? :)

File: site/content/en/docs/components/core/tide/config.md
Comment (kb000) [inline site/content/en/docs/components/core/tide/config.md:163]:
```suggestion
For a full list of properties of queries, please refer to [prow-config-documented.yaml](https://github.com/kubernetes-sigs/prow/blob/db89760fea406dd2813e331c3d52b53b5bcbd140/pkg/config/prow-config-documented.yaml#L1236).
```

File: cmd/checkconfig/main.go
Comment (jihoon-seo) [reply cmd/checkconfig/main.go]: I have updated `checkconfig` testcode in d3b59c2 to suppress error, because
IMO the testcode should be updated not to return error which occurs when the regexp `[^/]k8s[^/]` matches the error string,
where the error string in this case is:

```
"the following enable strict branchprotection context requirements even though Tide handles merges: [org: kubernetes]. See #5: https://docs.prow.k8s.io/docs/components/core/tide/maintainers/#best-practices Also note that this validation is imperfect, see the check-config code for details"
```

Please take a look.

**Comment by airbornepony** [comment]
Does anyone reviewing this PR believe they are in a position to approve soon?

In the future, it would be more efficient to contribute smaller PRs, focused on specific repo areas (e.g. code vs. site) as well as on the various topics that have been addressed. Thanks!

**Comment by jihoon-seo** [comment]
@airbornepony I see. You are right.
Next time I will open smaller PRs. And if needed, I may exclude code changes from this PR.

File: cmd/checkconfig/main.go
Comment (airbornepony) [reply cmd/checkconfig/main.go]: Documentation links should land at the Prow docs site (https://docs.prow.k8s.io), not the github source .md files, since the docs site provide a far superior user experience.

If we need to improve tests in some way to allow this, we should take that course.

**Comment by airbornepony** [comment]
@jihoon-seo Since nobody has responded affirming their ability to approve, feel free to assign to me, and I'll take the time to get this done.

Reviewers: If you have important concerns, please comment. Otherwise, let's avoid any nit/trivial comments in order to merge this PR in a timely way. Small follow-up PRs are always welcome. Thanks all!

**Comment by petr-muller** [comment]
I'm happy to stamp `approved` on PRs that get reviewed by others but don't have capacity to perform full reviews right now

**Comment by petr-muller** [comment]
I agree that we don't need to be perfect, I have no problem with letting improvements like this merge with possible issues fixed as a followup.

**Comment by jihoon-seo** [comment]
@airbornepony as per your comment, let me assign this PR to you:
/assign @airbornepony
Please take a look. Thank you!

**Review by airbornepony — CHANGES_REQUESTED** [review]
Nice work. I only found a couple of errors. There is one nit suggested by others that is okay to include in a final commit. Then we should be ready to go.

**Review by airbornepony — APPROVED** [review]
LGTM. This was a huge effort, thanks for doing this!

**Comment by jihoon-seo** [comment]
@cjwagner @petr-muller @kb000 @airbornepony Thank you all for your reviews and comments!

**Pull request merged**
